packages/client/src/proxy.js

```diff
@@ -219,11 +237,31 @@ export function proxyAgentFor(url, options) {
   let { protocol, hostname } = new URL(url);
   let cachekey = `${protocol}//${hostname}`;
 
-  if (!cache.has(cachekey)) {
-    cache.set(cachekey, protocol === 'https:'
-      ? new ProxyHttpsAgent(options)
-      : new ProxyHttpAgent(options));
+  // If we already have a cached agent, return it
+  if (cache.has(cachekey)) {
+    return cache.get(cachekey);
   }
 
-  return cache.get(cachekey);
+  try {
+    let agent;
+    const pacUrl = process.env.PERCY_PAC_FILE_PATH;
+
+    // If PAC URL is provided, use PAC proxy
+    if (pacUrl) {
+      logger('client:proxy').info(`Using PAC file from: ${pacUrl}`);
+      agent = createPacAgent(stripQuotesAndSpaces(pacUrl), options);
```

Comment:
```suggestion
      agent = createPacAgent(pacUrl, options);
```
Move that inside the `createPacAgent` function